src/dbally/iql_generator/iql_generator.py

```diff
@@ -60,8 +67,12 @@ async def generate_iql(
         prompt_format = IQLGenerationPromptFormat(
             question=question,
             filters=filters,
+            aggregations=aggregations,
             examples=examples,
         )
+        if aggregations and not filters:
```

Comment:
So the same `IQLGenerator` class is used for two separate functionalities in the system with different prompts and the decision which one to execute depends on whether the "filters" attribute is present (and the list is non-empty) or not? That seems confusing to me and this behavior is not explained in the documentation of this class.

I believe those should be split into two separate classes. Or, at the very least, the switch of behavior should be very explicitly described in the docstring for the class itself and the "aggregations" and "filters" arguments.